<!DOCTYPE html>
<html lang="en">
<title>Materials</title>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Poppins">
<style>
body,h1,h2,h3,h4,h5 {font-family: "Poppins", sans-serif}
body {font-size:16px;}
.w3-half img{margin-bottom:-6px;margin-top:16px;opacity:0.8;cursor:pointer}
.w3-half img:hover{opacity:1}
img {
    display: block;
    margin-left: auto;
    margin-right: auto;
}
</style>
<body>

<!-- Sidebar/menu -->
<nav class="w3-sidebar w3-green w3-collapse w3-top w3-large w3-padding" style="z-index:3;width:300px;font-weight:bold;" id="mySidebar"><br>
  <a href="javascript:void(0)" onclick="w3_close()" class="w3-button w3-hide-large w3-display-topleft" style="width:100%;font-size:22px">Close Menu</a>
  <div class="w3-container">
    <h3 class="w3-padding-64"><b>Supplementary Materials</b></h3>
  </div>
  <div class="w3-bar-block">
    <a href="#" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-white">Summary</a>
    <a href="#data" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-white">Datasets</a>
    <a href="#software" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-white">Software and Workflows</a>
    <a href="#docker" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-white">Docker Image</a>
    <a href="#prov" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-white">Provenance Traces</a>
    <a href="#bib" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-white">Bibliography</a>
    <a href="#authors" onclick="w3_close()" class="w3-bar-item w3-button w3-hover-white">About the authors</a>
  </div>
</nav>

<!-- Top menu on small screens -->
<header class="w3-container w3-top w3-hide-large w3-green w3-xlarge w3-padding">
  <a href="javascript:void(0)" class="w3-button w3-green w3-margin-right" onclick="w3_open()">☰</a>
  <!--<span>Company Name</span>-->
</header>

<!-- Overlay effect when opening sidebar on small screens -->
<div class="w3-overlay w3-hide-large" onclick="w3_close()" style="cursor:pointer" title="close side menu" id="myOverlay"></div>

<!-- !PAGE CONTENT! -->
<div class="w3-main" style="margin-left:340px;margin-right:40px">

  <!-- Header -->
  <div class="w3-container" style="margin-top:80px" id="showcase">
    <h1 ><b>Supplementary materials for paper: Computational challenges to reproducibility, robustness, extensibility and reuse in multi-omics: a meta-workflow-based case study (Under review)</b></h1>
    <!--<h2 class="w3-text-green"><b></b></h2>
    <hr style="width:50px;border:5px solid green" class="w3-round">-->
  </div>


  <div class="w3-container" id="summary" style="margin-top:15px">
    <!--<h1 class="w3-xxxlarge w3-text-green"><b>Services.</b></h1>-->
    <hr style="width:50px;border:5px solid green" class="w3-round">
    <p>This page provides pointers to the materials (datasets, software, workflows, provenance traces and Docker image) used in a paper currently <strong>under review</strong> (more details about the publication will be announced at a later stage)</p>

    <p><strong>Summary</strong>: The primary goal of our study is to demonstrate an example computational paradigm that uses meta-workflows (i.e., workflows that assess the results of two or more workflows) alongside abstract components (i.e, components that can be implemented with multiple tools) to overcome barriers facing multi-omic discovery.  We characterize the reproducibility of multi-omics discoveries by attempting a figure-for-figure reproduction of the seminal paper by <a href="#bib">Zhang et al [1]</a>.  We then assess the robustness of its findings by re-running all the experiments with newly acquired data, discussing the differences of the obtained results.</p>

    <p>This page and the materials described on it (excluding external references) is available in <a href="https://zenodo.org/badge/latestdoi/350090124">Zenodo under DOI 10.5281/zenodo.4633542</a>. </p>
  </div>


  <div class="w3-container" id="data" style="margin-top:75px">
    <h1 class="w3-xxxlarge w3-text-green"><b>Datasets.</b></h1>
    <hr style="width:50px;border:5px solid green" class="w3-round">
    <h3>Input Datasets</h3>
    <p>Note that availability on some of the following datasets may be subject to the the platforms where they are stored: </p>
    <ul>
      <li><a href="https://cptac-data-portal.georgetown.edu/cptac/s/S027">Proteomic Analysis of Colon and Rectal Carcinoma Using Standard and Customized Databases</a></li>
      <li><a href=" http://doi.org/10.5281/zenodo.4633304">Custom and Standard reference sequence databases</a> (including custom_merged_OV.fasta, custom_merged_BRCA.fasta, custom_merged_CRC95.fasta as custom datasets; and  hg19_refseq_pro_seq.fasta, humanRefSeq_Version54_with_tryp_DECOY.fasta as reference datasets)</li>
      <li><a href="https://cptac-data-portal.georgetown.edu/cptac/s/S022">MzML files corresponding to 95 samples were obtained from CPTAC</a></li>
      <li><a href="https://www.proteomicsdb.org/proteomicsdb/#projects/35/256">RAW files for 41 samples from the NCI-60 full proteome data were downloaded from ProteomicsDB</a></li>
      <li><a href="https://cptac-data-portal.georgetown.edu/cptac/s/S026">Proteomics data for 174 ovarian cancer samples was obtained from CPTAC</a></li>
      <li><a href="https://cptac-data-portal.georgetown.edu/cptac/s/S029">37 mzML files corresponding to 108 Breast cancer samples (including three duplicates and three normals from global proteome data)</a></li>
    </ul>

    <!--
    <h3>Outputs</h3>
    <ul>
      <li>Point to each here (as in paper)</li>
    </ul>
  -->

  </div>

  <div class="w3-container" id="software" style="margin-top:75px">
    <h1 class="w3-xxxlarge w3-text-green"><b>Software and Workflows.</b></h1>
    <hr style="width:50px;border:5px solid green" class="w3-round">
    The pointers for using the main software and workflows can be found below:
    <ul>
      <li><a href="software-packages.md">List of the software packages</a> (and corresponding version) installed in the Docker image</li>
      <li><a href="workflow_files_description.pdf">Documentation of the input and output files used by each workflow</a></li>
      <li><a href="worflow_parameters_description.pdf">Documentation of the parameters used for each workflow</a></li>
      <li><a href="code.zip">List of the code wrappers used to create workflow components for all workflows</a>. You can also browse them <a href="https://github.com/KnowledgeCaptureAndDiscovery/wings-genomics/tree/main/code/library">on GitHub</a>. (Note: these wrappers do not contain the dependencies required for each tool, they are ready to use in the Docker image)</li>
      <li><a href="workflows/wf-figures.pdf">Document with an overview of the workflows used in the manuscript</a>. The workflows themselves can be found within the Docker image (in WINGS), or by downloading the <a href="workflows.zip">workflow specification files</a> (which contain all the implementation details). If you want to explore the contents of the zip files, you can also browse the <a href="https://github.com/KnowledgeCaptureAndDiscovery/wings-genomics/tree/main/workflows">on GitHub</a>. (Note: data is not included in these files)</li>
    </ul>
  </div>

  <div class="w3-container" id="docker" style="margin-top:75px">
    <h1 class="w3-xxxlarge w3-text-green"><b>Docker image.</b></h1>
    <hr style="width:50px;border:5px solid green" class="w3-round">

    <ul>
      <li>
        A Docker image with WINGS and sample data, software and workflows is available on <a href="http://doi.org/10.5281/zenodo.4612080">Zenodo</a>
      </li>
      <li>
        The Docker image is currently available in DockerHub with the id: <a href="https://hub.docker.com/r/kcapd/wings-genomics-full/tags?page=1&ordering=last_updated">kcapd/wings-genomics-full:latest</a> (availability may be subject to DockerHub policies)
      </li>
      <li>
        The instructions for downloading and running the Docker image locally can be found on <a href="docker-instructions.md">GitHub</a>.
      </li>
    </ul>
  </div>

  <div class="w3-container" id="prov" style="margin-top:75px">
    <h1 class="w3-xxxlarge w3-text-green"><b>Provenance traces.</b></h1>
    <hr style="width:50px;border:5px solid green" class="w3-round">

    Below you can find provenance traces for the experiments run in the paper, following the <a href="http://www.opmw.org/model/OPMW/">OPMW-PROV model</a> (and extending the <a href="https://www.w3.org/TR/prov-o/">W3C PROV standard</a>). Note that the pointers to code and data are relative URLs, which won't resolve (they indicate the names of the files that have been used, but the paths are local to the Docker image). The code components and data referred to can be found in the dataset and code sections.
    <ul>
      <li><a href="provenance/MultiProteomic_S-d4df9090-67c0-4e01-99dc-43654df14c15">Proteomic Run (Stage 3 reanalysis)</a>. Also browsable <a href="
  https://www.opmw.org/export/omics/page/resource/WorkflowExecutionAccount/MultiProteomic_S-d4df9090-67c0-4e01-99dc-43654df14c15">online</a></li>
      <li><a href="provenance/SingleProteogeno-333b42f6-bd36-47ad-9b6f-9c2f8e0f0d55">Proteogenomic Run</a>. Also browsable <a href="https://www.opmw.org/export/omics/page/resource/WorkflowExecutionAccount/SingleProteogeno-333b42f6-bd36-47ad-9b6f-9c2f8e0f0d55">online</a></li>
      <li><a href="provenance/SingleProteomic_-c00add8a-4c33-42dd-be7d-9523f05959be
">Pepitome Run (Figure 3)</a>. Also browsable <a href="https://www.opmw.org/export/omics/resource/WorkflowExecutionAccount/SingleProteomic_-c00add8a-4c33-42dd-be7d-9523f05959be
">online</a></li>
      <li><a href="provenance/xTandemSearch-67-38e22c05-4260-428e-9001-c3a2ad98de02">Xtandem Run (Figure 3)</a>. Also browsable <a href="https://www.opmw.org/export/omics/page/resource/WorkflowExecutionAccount/xTandemSearch-67-38e22c05-4260-428e-9001-c3a2ad98de02">online</a>.</li>
    </ul>

  </div>

  <div class="w3-container" id="bib" style="margin-top:75px">
    <h1 class="w3-xxxlarge w3-text-green"><b>Bibliography.</b></h1>
    <hr style="width:50px;border:5px solid green" class="w3-round">
    <p id ="zhang">[1] Zhang, B. et al. <strong>Proteogenomic characterization of human colon and rectal cancer</strong>. Nature 513, 382-387, doi:<a href="https://dx.doi.org/10.1038/nature13438">10.1038/nature13438</a> (2014)</p>
    </ul>
  </div>

  <!-- The Team -->
  <div class="w3-container" id="authors" style="margin-top:75px">
  <h1 class="w3-xxxlarge w3-text-green"><b>About the authors.</b></h1>
  <hr style="width:50px;border:5px solid green" class="w3-round">
  <div class="w3-row-padding">
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
          <img src="images/ravali.jpg" alt="Ravali Adusumilli" style="width:100px;padding-top: 10px;">
        <div class="w3-container">
          <h3><a href="http://mallicklab.stanford.edu/peeps">Ravali Adusumilli</a></h3>
          <p class="w3-opacity">Researcher</p>
          <p>Bioinformaticist at the Mallik Lab of Stanford Univesity. She is interested in developing tools and pipelines for multi-omic analysis. </p>
        </div>
      </div>
    </div>
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
        <img src="images/dani.png" alt="Daniel Garijo" style="width:90px;padding-top: 10px;">
        <div class="w3-container">
          <h3><a href="https://dgarijo.com">Daniel Garijo</a></h3>
          <p class="w3-opacity">Researcher</p>
          <p>Researcher at the Polytechnic University of Madrid and past researcher at the Information Sciences Institute of the
University of Southern California. Daniel's research activities focus on
e-Science and the Semantic web, specifically on how to increase the
understandability of software and scientific workflows using provenance, metadata,
intermediate results and Linked Data. </p>
        </div>
      </div>
    </div>

  </div>

  <div class="w3-row-padding">
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
          <img src="images/varun.jpg" alt="Varun Ratnakar" style="width:100px;padding-top: 10px;">
        <div class="w3-container">
          <h3>Varun Ratnakar</h3>
          <p class="w3-opacity">Research Programmer</p>
          <p>Research programmer at the Information Sciences Institute of the University of Southern California. Ratnakar is the main developer of the <a href="https://wings-workflows.org/">WINGS workflow system</a>. </p>
        </div>
      </div>
    </div>
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
        <img src="images/unknown.png" alt="Arunima Srivastava" style="width:90px;padding-top: 10px;">
        <div class="w3-container">
          <h3>Arunima Srivastava</h3>
          <p class="w3-opacity">Data Scientist</p>
          <p>Research collaborator at the Mallick lab, Stanford. Arunima currently works as a Data Scientist at a major pharmaceutical company. </p>
        </div>
      </div>
    </div>

  </div>

  <div class="w3-row-padding">
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
          <img src="images/matt.jpeg" alt="Matt Chambers" style="width:100px;padding-top: 10px;">
        <div class="w3-container">
          <h3>Matt Chambers</h3>
          <p class="w3-opacity">Researcher</p>
          <p>Research collaborator at the Vanderbilt University, now working as a Bioinformatics and proteome informatics consultant. </p>
        </div>
      </div>
    </div>
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
        <img src="images/jing.jpg" alt="Jing Wang" style="width:80px;padding-top: 10px;">
        <div class="w3-container">
          <h3>Jing Wang</h3>
          <p class="w3-opacity">Research Assistant Professor</p>
          <p>Research Assistant Professor of Biomedical Informatics in the School of Medicine at Vanderbilt University. His research interests focus on integrating multi-dimensional omics data to understand cancer mechanism. His current work focuses on assessing biological relevance of mRNA and protein profiling data and developing computational tools to discover and interpret novel associations among cancer omics data.</p>
        </div>
      </div>
    </div>

  </div>

  <div class="w3-row-padding">
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
          <img src="images/xiaojing.jpg" alt="Xiaojing Wang" style="width:100px;padding-top: 10px;">
        <div class="w3-container">
          <h3>Xiaojing Wang</h3>
          <p class="w3-opacity">Research Instructor</p>
          <p>Research instructor of biomedical informatics in the School of Medicine at Vanderbilt University. Dr. Wang has extensive experience with biological data, including sequencing data analysis and gene expression. Her research focuses on using such resources in parallel with shotgun proteomics data, an emerging field termed proteogenomics, to understand cancer biology. Her current projects include developing bioinformatics methods for proteogenomics studies and its application to human cancer studies. </p>
        </div>
      </div>
    </div>
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
        <img src="images/david.jpg" alt="David L. Tabb" style="width:100px;padding-top: 10px;">
        <div class="w3-container">
          <h3>David L. Tabb</h3>
          <p class="w3-opacity">Professor</p>
          <p> Professor at the South African Tuberculosis Bioinformatics Initiative. Dr Tabb and his team have created a complete software pipeline for proteome informatics, contributing to the field-standard ProteoWizard library for mass spectrometry informatics, creating three distinct peptide identification algorithms, and developing the highly scalable IDPicker protein assembly environment. Dr Tabb moved to Stellenbosch University in 2015 to speed the development of proteomics in South Africa and to add support to tuberculosis researchers throughout Cape Town.</p>
        </div>
      </div>
    </div>

  </div>

  <div class="w3-row-padding">
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
          <img src="images/bingz.jpeg" alt="Bing Zhang" style="width:100px;padding-top: 10px;">
        <div class="w3-container">
          <h3><a href="https://www.bcm.edu/people-search/bing-zhang-33575">Bing Zhang</a></h3>
          <p class="w3-opacity">Professor</p>

          <p>Professor at the Lester and Sue Smith Breast Center of the Baylor College of Medicine (Houston, TX); and professor at the  Department of Molecular and Human Genetics of the  Baylor College of Medicine. Professor Zhang's research  interests focus on developing computational and statistical approaches that help translating multidimensional omics data into biological and clinical insights.  </p>
        </div>
      </div>
    </div>
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
        <img src="images/machiraju.jpg" alt="Raghu Machiraju" style="width:80px;padding-top: 10px;">
        <div class="w3-container">
          <h3>Raju Machiraju</h3>
          <p class="w3-opacity">Professor</p>
          <p>TDAI Chief Data Scientist; Professor of Computer Science and Engineering at the College of Engineering and Biomedical Informatics, College of Medicine. Dr. Machiraju’s interests include visual analytics, modeling, and machine learning, especially as they apply to topics in biology, medicine, and engineering. Over the years he has been working increasingly on problems of computational biology and bioinformatics including the discovery of biomarkers and disease subtypes and the modeling and reconstruction of signal transduction networks.
</p>
        </div>
      </div>
    </div>

  </div>

  <div class="w3-row-padding">
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
          <img src="images/gil.jpg" alt="Yolanda Gil" style="width:100px;padding-top: 10px;">
        <div class="w3-container">
          <h3><a href="http://www.isi.edu/~gil/">Yolanda Gil</a></h3>
          <p class="w3-opacity">Research Professor</p>
          <p>Director of Knowledge Technologies and at the Information Sciences Institute of the University of Southern California, and Research Professor in the Computer Science Department. Her research interests include intelligent user interfaces, social knowledge collection, provenance and assessment of trust, and knowledge management in science. Dr Gil's most recent work focuses on intelligent workflow systems to support collaborative data analytics at scale.</p>
        </div>
      </div>
    </div>
    <div class="w3-col m6 w3-margin-bottom">
      <div class="w3-light-grey">
        <img src="images/parag.jpg" alt="Parag Mallick" style="width:120px;padding-top: 10px;">
        <div class="w3-container">
          <h3><a href="https://med.stanford.edu/profiles/parag-mallick">Parag Mallick</a></h3>
          <p class="w3-opacity">Associate Professor in Radiology</p>
          <p>Assistant professor (Research) in Radiology. Dr. Mallick is also member of the Stanford Cancer Institute and a faculty fellow of Stanford ChEM-H. After completing his PhD, he trained with Ruedi Aebersold in clinical proteomics and systems biology at the Institute for Systems Biology. </p>
        </div>
      </div>
    </div>

  </div>

  </div>

<!-- End page content -->
</div>

<!-- W3.CSS Container -->
<div class="w3-light-grey w3-container w3-padding-32" style="margin-top:75px;padding-right:58px"><p class="w3-right">Designed deived from <a href="https://www.w3schools.com/w3css/default.asp" title="W3.CSS" target="_blank" class="w3-hover-opacity">w3.css</a></p></div>

<script>
// Script to open and close sidebar
function w3_open() {
  document.getElementById("mySidebar").style.display = "block";
  document.getElementById("myOverlay").style.display = "block";
}

function w3_close() {
  document.getElementById("mySidebar").style.display = "none";
  document.getElementById("myOverlay").style.display = "none";
}

// Modal Image Gallery
function onClick(element) {
  document.getElementById("img01").src = element.src;
  document.getElementById("modal01").style.display = "block";
  var captionText = document.getElementById("caption");
  captionText.innerHTML = element.alt;
}
</script>

</body>
</html>
